### PR TITLE
Implement static set/get items on records with integer index

### LINKED ISSUE #6655: Record staticgetitem with integer index not implemented.
## Reporting a bug

- [x] I have tried using the latest released version of Numba (most recent is
 visible in the change log (https://github.com/numba/numba/blob/master/CHANGE_LOG).
- [x] I have included a self contained code sample to reproduce the problem.
  i.e. it's possible to run as 'python bug.py'.

Context: https://numba.discourse.group/t/accessing-structured-array-scalars-by-index/446

This ought to work:
```python
from numba import njit
import numpy as np
arr = np.array([(1, 2., 3.)], dtype='i, f, f')

@njit
def get_elem(arr, idx):
    return arr[idx][0] # record index is const 0

print(get_elem(arr, 0))
```
but doesn't as it is not implemented.

This does work:
```python
from numba import njit
import numpy as np
arr = np.array([(1, 2., 3.)], dtype='i, f, f')

@njit
def get_elem(arr, idx):
    return arr[idx]['f0'] # record index is const 'f0'

print(get_elem(arr, 0))
```
seems strangely inconsistent!

### PR DESCRIPTION
**Pull request by stuartarchibald**
As title.

Closes #6655

### CODE REVIEW

File: numba/core/typing/arraydecl.py [hunk 1]
```diff
@@ -562,6 +562,14 @@ def generic(self, args, kws):
             ret = record.typeof(idx.literal_value)
             assert ret
             return signature(ret, *args)
+        if isinstance(record, types.Record) and isinstance(idx, types.IntegerLiteral):
```
Comment (esc) [inline numba/core/typing/arraydecl.py:565]:
```suggestion
        elif isinstance(record, types.Record) and isinstance(idx, types.IntegerLiteral):
```

To clarify that it is either String or Integer?

File: numba/core/typing/arraydecl.py [hunk 2]
```diff
@@ -575,6 +583,14 @@ def generic(self, args, kws):
             expectedty = record.typeof(idx)
             if self.context.can_convert(value, expectedty) is not None:
                 return signature(types.void, record, types.literal(idx), value)
+        if isinstance(record, types.Record) and isinstance(idx, int):
```
Comment (esc) [inline numba/core/typing/arraydecl.py:586]: As above.

**Review by esc — CHANGES_REQUESTED** [review]
Only one statement to consider.

File: numba/core/typing/arraydecl.py (hunk 1 above)
Comment (stuartarchibald) [reply numba/core/typing/arraydecl.py]: Thanks, in 9d08da6 I've made this an `elif` and factored out the instance check on the `types.Record` to save testing twice!

Comment (esc) [reply numba/core/typing/arraydecl.py]: yeah, that's good!

**Review by esc — APPROVED** [review]
Thanks for the patch!

**Comment by stuartarchibald** [comment]
> Thanks for the patch!

Thanks for the review.

**Pull request merged**